storage/shard.go

```diff
@@ -253,7 +257,6 @@ func (s *dbShard) IsBlockRetrievable(blockStart time.Time) bool {
 	}
 	panic(fmt.Errorf("shard queried is retrievable with bad flush state %d",
 		flushState.Status))
-	return false
```

Comment:
Interesting. My vscode linter complains when I have this. 
![screen shot 2017-06-30 at 17 40 20](https://user-images.githubusercontent.com/199982/27754989-3838fdc6-5dbb-11e7-852d-3d15ccfb38ea.png)